recipes/cubicinterpolation/all/conanfile.py

```diff
@@ -0,0 +1,56 @@
+from conans import ConanFile, CMake, tools
+import glob
+import os
+
+class CubicInterpolationConan(ConanFile):
+    name = "CubicInterpolation"
+    homepage = "https://github.com/MaxSac/cubic_interpolation"
+    version = "0.1"
+    license = "MIT"
+    author = "Maximilian Sackel mail@maxsac.de"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Leightweight interpolation library based on boost and eigen."
+    topics = ("interpolation", "splines", "cubic", "bicubic", "boost", "eigen3")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake_find_package"
```

Comment:
```suggestion
    generators = "cmake"
```

Cmake generator is required for conan_basic_setup

Reply:
To make the cmake generator work i had to modify the cmake lists slightly done in this [commit](https://github.com/MaxSac/cubic_interpolation/commit/b355c90ec60be53a992c03c90aebc6e8cf27733a). 
I created a new bug fix release and updated the [conandata.yml](https://github.com/conan-io/conan-center-index/blob/42f7675cd3723b20a458b0cbbae250a12a800664/recipes/cubicinterpolation/all/conandata.yml).

Reply:
@uilianries is it allowed to use the cmake_find_package generator if this is considered consistently in the project?
This would allow not forcing people to install conan, which is the biggest negative argument when people already have all the dependencies. 

Sorry for the late questions, but this is something you only come across when more and more people are using it. :)

Reply:
Yes, it's allowed. However, it's about package generation, you must have Conan installed. For consumer they also need Conan for package installing.